src/interface.cpp

```diff
@@ -264,6 +266,53 @@ static NAN_METHOD(transactionHash) {
   info.GetReturnValue().Set(ret);
 }
 
+static NAN_METHOD(bundleMiner) {
+  uint64_t index = 0;
+
+  if (info.Length() < 6) {
+    Nan::ThrowError("Wrong number of arguments");
+    return;
+  }
+
+  if (!info[0]->IsArray() || !info[1]->IsNumber() || !info[2]->IsArray() ||
+      !info[3]->IsNumber() || !info[4]->IsNumber() || !info[5]->IsNumber()) {
+    Nan::ThrowError("Wrong arguments");
+    return;
+  }
+
+  byte_t bundleNormalizedMax[81];
+  v8::Local<v8::Array> bundle_array = v8::Local<v8::Array>::Cast(info[0]);
+  for (size_t i = 0; i < bundle_array->Length(); i++) {
+    bundleNormalizedMax[i] =
+        bundle_array->Get(i)->NumberValue(Nan::GetCurrentContext()).FromJust();
+  }
+
+  uint64_t security =
+      static_cast<uint64_t>(Nan::To<unsigned>(info[1]).FromJust());
+  uint64_t essenceLength =
+      static_cast<uint64_t>(Nan::To<unsigned>(info[3]).FromJust());
+  trit_t *essence = (trit_t *)malloc(sizeof(trit_t) * essenceLength);
+  v8::Local<v8::Array> essence_array = v8::Local<v8::Array>::Cast(info[2]);
+  for (size_t i = 0; i < essence_array->Length(); i++) {
+    essence[i] =
+        essence_array->Get(i)->NumberValue(Nan::GetCurrentContext()).FromJust();
+  }
+  uint64_t count = static_cast<uint64_t>(Nan::To<unsigned>(info[4]).FromJust());
+  uint64_t nprocs =
+      static_cast<uint64_t>(Nan::To<unsigned>(info[5]).FromJust());
+
+  if (bundle_miner_mine(bundleNormalizedMax, security, essence, essenceLength,
+                        count, nprocs, &index) != RC_OK) {
+    info.GetReturnValue().Set(-1);
+    free(essence);
+    Nan::ThrowError("Bundle mining failed");
```

Comment:
If an error is thrown, will we still receive `-1` as the index?

Reply:
No idea :)

Reply:
I wonder if we should just return `-1` and not throw an error since that's what we've done for Android and iOS